src/scripts/generate_binary_ops.py

```diff
@@ -0,0 +1,305 @@
+#!/usr/bin/env python
+
+# Copyright JS Foundation and other contributors, http://js.foundation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import sys
+import argparse
+import collections
+import random
+import math
+import subprocess
+
+# Local packages
+import settings
+path = os.path.join(settings.TEMPLATE_DIR)
+sys.path.append(path)
+import validate # Function
+from constants import _constants # Constant values
+from parse_args import parse_args # Argument parser
+from util import unsigned32 # Number to unsigned int 32 bit
+from generator_base import Generator # Base class
+from utils import _write_file # File writer
+
+# Class for expressions
+class GenSequence(Generator):
+
+    _GENERATED_FILENAME = '''binary-ops-{NUMBER}.js''' # The file name prefix of the generated tests
```

Comment:
This constant will be passed to the file_write, in a setting object. How about declaring the settings object here?

Reply:
It is not possible, because there is no way to see `self.options.output` from the inner class, because it is in the outer class.